samtranslator/model/sam_resources.py

```diff
@@ -51,6 +53,11 @@
 from samtranslator.model.role_utils import construct_role_for_resource
 from samtranslator.model.xray_utils import get_xray_managed_policy_name
 
+# max_logical_id_length + max_unique_id_length + len(prefix) must be less than or equal to 22
```

Comment:
Is there any reference to the number 22?

Reply:
updated documentation in code for better understanding